python/packages/kagent-langgraph/src/kagent/langgraph/_executor.py

```diff
@@ -171,12 +202,245 @@ async def _stream_graph_events(
                 )
             )
 
+    async def _handle_interrupt(
+        self,
+        interrupt_data: list[Any],
+        task_id: str,
+        context_id: str,
+        event_queue: EventQueue,
+        task_store: TaskStore,
+    ) -> None:
+        """Handle interrupt from LangGraph and convert to A2A input_required event."""
+        # Extract interrupt details
+        # Format: [{"value": {"action_requests": [...], "review_configs": [...]}}]
+        if not interrupt_data:
+            logger.warning("Empty interrupt data received")
+            return
+
+        # Safely extract interrupt value
+        first_item = interrupt_data[0]
+        if hasattr(first_item, "value"):
+            interrupt_value = first_item.value
+        elif isinstance(first_item, dict):
+            interrupt_value = first_item
+        else:
+            logger.error(f"Unexpected interrupt data type: {type(first_item)}")
+            return
+
+        action_requests = interrupt_value.get("action_requests", [])
+        review_configs = interrupt_value.get("review_configs", [])
+
+        # Build approval message for user
+        parts = []
+
+        # Add header
+        parts.append(Part(TextPart(text="**Approval Required**\n\n")))
+        parts.append(Part(TextPart(text="The following actions require your approval:\n\n")))
+
+        # List each action
+        for action in action_requests:
+            tool_name = action.get("name", "unknown")
+            tool_args = action.get("args", {})
+
+            # Escape backticks in tool name to prevent markdown breaking
+            escaped_tool_name = str(tool_name).replace("`", "\\`")
+            parts.append(Part(TextPart(text=f"**Tool**: `{escaped_tool_name}`\n")))
+            parts.append(Part(TextPart(text="**Arguments**:\n")))
+            for key, value in tool_args.items():
+                # Escape backticks in values to prevent markdown breaking
+                escaped_key = str(key).replace("`", "\\`")
```

Comment:
extract the backticks escape to a separate func since it's used multiple times here